network/network_test.go

```diff
@@ -742,91 +739,122 @@ func TestNetwork_Reprocess(t *testing.T) {
+		setup, eventManager := newSetup(t)
+		setup.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return([]dag.Transaction{tx}, nil)
+		setup.state.EXPECT().ReadPayload(gomock.Any(), tx.PayloadHash()).Return([]byte("payload"), nil)
 
-		ctx.network.Reprocess("application/did+json")
+		messages := subscribe(ctx, t, eventManager)
 
-		test.WaitFor(t, func() (bool, error) {
-			foundMutex.Lock()
-			defer foundMutex.Unlock()
-			return found > 0, nil
-		}, 100*time.Millisecond, "timeout waiting for event")
+		_, err := setup.network.ReprocessContentType(ctx, "application/did+json")
+		if err != nil {
+			t.Error("reprocess error:", err)
+		}
+
+		if len(messages) == 0 {
+			t.Error("subscription received no messages")
+		}
 	})
 
-	t.Run("ignores other transactions", func(t *testing.T) {
-		ctx, eventManager := setup(t)
-		ctx.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return([]dag.Transaction{tx}, nil)
-		ctx.state.EXPECT().ReadPayload(gomock.Any(), tx.PayloadHash()).Return([]byte("payload"), nil)
+	t.Run("mismatch", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
 
-		var found int
-		subscribe(&found, eventManager)
+		setup, eventManager := newSetup(t)
+		setup.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return([]dag.Transaction{tx}, nil)
+		setup.state.EXPECT().ReadPayload(gomock.Any(), tx.PayloadHash()).Return([]byte("payload"), nil)
 
-		ctx.network.Reprocess("application/did+vc")
+		messages := subscribe(ctx, t, eventManager)
 
-		test.WaitForNoFail(t, func() (bool, error) {
-			foundMutex.Lock()
-			defer foundMutex.Unlock()
-			return found > 0, nil
-		}, 100*time.Millisecond)
+		_, err := setup.network.ReprocessContentType(ctx, "application/did+vc")
+		if err != nil {
+			t.Error("reprocess error:", err)
+		}
 
-		assert.Equal(t, 0, found)
+		if n := len(messages); n != 0 {
+			t.Errorf("received %d messages, want 0", n)
+		}
 	})
 
-	t.Run("stops on error", func(t *testing.T) {
-		ctx, eventManager := setup(t)
-		ctx.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return(nil, errors.New("b00m!"))
+	t.Run("error", func(t *testing.T) {
+		t.Run("query", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+			defer cancel()
 
-		var found int
-		subscribe(&found, eventManager)
+			testErr := errors.New("test error")
 
-		ctx.network.Reprocess("application/did+vc")
+			setup, eventManager := newSetup(t)
+			setup.state.EXPECT().FindBetweenLC(gomock.Any(), uint32(0), uint32(1000)).Return(nil, testErr)
 
-		test.WaitForNoFail(t, func() (bool, error) {
-			foundMutex.Lock()
-			defer foundMutex.Unlock()
-			return found > 0, nil
-		}, 100*time.Millisecond)
+			messages := subscribe(ctx, t, eventManager)
 
-		assert.Equal(t, 0, found)
-	})
+			_, err := setup.network.ReprocessContentType(ctx, "application/did+vc")
+			switch {
```

Comment:
This switch would have been a single assert when using testify

Reply:
True. The message printed on either case is terrible though.